### ISSUE TITLE
How to make a text input auto active ?

### ISSUE THREAD

**Issue by jbtrystram** [patternfly/patternfly-react#6676]
When creating a simple modal with patternfly, I'd like the field in the form to be active (i.e. I don't need to click in the field before typing text). 
I did not find how to do that on the patternfly website. 
I want to have the same behaviour as on github for example : when you open an issue, the "title" field is automatically selected. 

Here is the code I use to create my modal : 
```
Formatted HTML:

<div class="pf-l-bullseye">
  <div aria-labelledby="modal-title" role="dialog" aria-describedby="modal-description" class="pf-m-sm pf-c-modal-box" aria-modal="true">
    <button class="pf-c-button pf-m-plain" aria-label="Close dialog" type="button"><i class="fas fa-times" aria-hidden="true"></i></button>
    <header class="pf-c-modal-box__header">
      <h1 class="pf-c-modal-box__title" id="modal-title-modal-with-form">Create an application</h1>
    </header>
    <div class="pf-c-modal-box__body">
      <p></p>
    </div>
    <div class="pf-c-modal-box__body">
      <form novalidate="novalidate" class="pf-c-form">
        <div class="pf-c-form__group">
          <div class="pf-c-form__group-label"></div>
          <div class="pf-c-form__group-control"><input placeholder="Application ID" class="pf-c-form-control" aria-invalid="true" id="" name="" type="text"></div>
        </div>
      </form>
    </div>
    <footer class="pf-c-modal-box__footer"><button disabled="disabled" class="pf-c-button pf-m-primary" type="button">Create</button></footer>
  </div>
</div>
```

**Comment by mcarrano** [comment]
@nicolethoen @tlabaj have either of you had a chance to look at this?

**Comment by nicolethoen** [comment]
@mcarrano no, but it was actually at the top of my list for something to hopefully address quickly this milestone

**Comment by nicolethoen** [comment]
@jbtrystram I have built a codesandbox to demonstrate how to put the focus on the name input when the modal opens: 
https://codesandbox.io/s/confident-tamas-z7bpo?file=/index.js

Let me know if that doesn't work for you.
@mcarrano @tlabaj should I update the Modal with form example to do this?

**Comment by mcarrano** [comment]
Are you talking about this example @nicolethoen ? https://www.patternfly.org/v4/components/modal#with-form

If so, yes, I think it would be helpful for this to be included in the example.

**Comment by jbtrystram** [comment]
@nicolethoen Thanks for building the example. 
I believe the relevant section that make the field active when the modal is opened is  : 
```
 React.useEffect(() => {
    if (isModalOpen) {
      if (nameInputRef && nameInputRef.current) {
        nameInputRef.current.focus();
      }
    }
  }, [isModalOpen, nameInputRef]);
  ```

I am writing my web application using Rust and WASM, not react, so I need to find out how to make the field focused with HTML/CSS, and i don't see any differents properties on the focused field vs other fields.

**Comment by nicolethoen** [comment]
@jbtrystram 
I do believe you will still need to use javascript to manipulate browser focus. Without using react, you could give the Text Input an `id` and then move the browser focus after the modal is triggered to open by doing something like `document.getElementById("myAnchor").focus();`

**Comment by jbtrystram** [comment]
@nicolethoen  okay, thanks for the tip !
